usr/sbin/regionsrv-enabler-azure

```diff
@@ -97,20 +97,27 @@ utils.start_logging()
 service_name = 'guestregister'
 license_type = get_license_type()
 
-if ('BYOS' in license_type and has_license_changed(license_type) and
-    not utils.uses_rmt_as_scc_proxy()):
+if ('BYOS' in license_type and not utils.uses_rmt_as_scc_proxy()):
+    # Registered to SCC
+    license_has_changed = has_license_changed(license_type)
+    if license_has_changed == False:
```

Comment:
If we are using `False` here then `has_license_changed()`  needs get a `return False` at the end of the function. 
```
>>> None == False
False
```
This code will fail when there is no state file because `has_license_changed()` will return `None`. Alternatively stick with `if not has_license_changed`

If there is no state file, i.e. `CACHE_LICENSE_PATH` does not exist then the system has not ever had a license setting in the metadata. If there is no license in the metadata we should not ever take any action because we cannot determine whether or not the instance is PAYG or BYOS only from looking at the metadata delivered by IMDS.

Ultimately this code is a state machine. It is probably good to put the states we need to handle into a doc string at the top of the file. That way it is easier in the future to correlate the implemented logic with the documented states we need to handle.

Reply:
> If we are using `False` here then `has_license_changed()` needs get a `return False` at the end of the function.
> 
> ```
> >>> None == False
> False
> ```
> 
> This code will fail when there is no state file because `has_license_changed()` will return `None`. Alternatively stick with `if not has_license_changed`
> 
> If there is no state file, i.e. `CACHE_LICENSE_PATH` does not exist then the system has not ever had a license setting in the metadata. If there is no license in the metadata we should not ever take any action because we cannot determine whether or not the instance is PAYG or BYOS only from looking at the metadata delivered by IMDS.

Fair point, the use of the re raise of the exception or `None` to determine if any de-registering action should be taken
when `BYOS` could have a better approach

> Ultimately this code is a state machine. It is probably good to put the states we need to handle into a doc string at the top of the file. That way it is easier in the future to correlate the implemented logic with the documented states we need to handle.

Will do

Reply:
Done